draft-jennings-moq-secure-objects.md

```diff
@@ -270,50 +293,31 @@ SECURE_OBJECT_AAD {
 }
 ~~~
 
+Serialized Immutable Extensions MUST include the `Secure Object KID header
+extension` containing the Key ID.
+
 ## Nonce Formation {#nonce}
 
 The Group ID and Object ID for an object are used to form a 96-bit counter (CTR)
 value, which XORed with a salt to form the nonce used in AEAD encryption.  The
-counter value is formed by encoding the Group ID and Object ID as QUIC varints,
-then concatenating these representations.  This scheme MUST NOT be applied to an
-object where group ID is larger than 2<sup>62</sup> or the object ID is larger
-than 2<sup>30</sup>.
-
-~~~  pseudocode
-
-def encode_varint(x):
-    if x < 0x40:
-        return (x, 8)
-    elif x < 0x4000:
-        return (0x4000 + x, 16)
-    elif x < 0x40000000:
-        return (0x80000000 + x, 32)
-    elif x < 0x4000000000000000:
-        return (0xc000000000000000 + x, 64)
-
-def encode_ctr(group_id, object_id):
-    (group_id, group_bits) = encode_varint(group_id)
-    (object_id, object_bits) = encode_varint(object_id)
+counter value is formed by bitwise concatenating the  Group ID as 64 bit integer
+and  Object ID as 32 bit integer. This scheme MUST NOT be applied to an object
+where group ID is larger than 2<sup>64</sup> or the object ID is larger
+than 2<sup>32</sup>.
 
-    group_shift = 96 - group_bits
-    object_shift = group_shift - object_bits
-
-    return (group_id << group_shift) | (object_id << object_shift)
-~~~
 
 ## Key Derivation {#keys}
 
 Encryption and decryption use a key and salt derived from the `track_base_key`
-associated with a Key ID.  Given a `track_base_key` value, the key and salt are derived
-using HMAC-based Key Derivation Function (HKDF) {{!RFC5869}} as follows:
+associated with a Key ID. Given a `track_base_key` value, the key and salt are
+derived using HMAC-based Key Derivation Function (HKDF) {{!RFC5869}} as follows:
 
 ~~~ pseudocode
-def derive_key_salt(Key ID, track_base_key):
+def derive_key_salt(Key ID, track_base_key, Serialized Full Track Name):
   moq_secret = HKDF-Extract("", track_base_key)
-  moq_key_label = "MOQ 1.0 Secret key " + Key ID + cipher_suite
+  moq_key_label = "MOQ 1.0 Secret key " + Key ID + Serialized Full Track Name + cipher_suite
   moq_key =
     HKDF-Expand(moq_secret, moq_key_label, AEAD.Nk)
-
   moq_salt_label = "MOQ 1.0 Secret salt " + Key ID + cipher_suite
```

Comment:
change this to 

moq_salt_label = "MOQ 1.0 Secret salt " + Serialized Full Track Name  + cipher_suite +  Key ID